src/components/TradeWidget/index.tsx

```diff
@@ -336,6 +340,25 @@ function calculateReceiveAmount(priceValue: string, sellValue: string): string {
   return receiveAmount
 }
 
+interface TokenAdderProps {
+  tokenAddress: string
+  networkId: number
+  onSelectChange: (selected: TokenDetails) => void
+}
+
+const TokenAdder: React.FC<TokenAdderProps> = ({ tokenAddress, networkId, onSelectChange }: TokenAdderProps) => {
```

Comment:
consider moving this to it's own file

Reply:
Does it make sense that instead of defining the hook in it's own file, we define the component TokenAdderModal with all the logic there?